catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessorTest.java

```diff
@@ -0,0 +1,360 @@
+    assertThat(detectedMgrsStrings, both(hasSize(1)).and(contains("4QFJ0012346789")));
+  }
+
+  @Test
+  public void testProcessorDetectsPrecision4NoSpace() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4Q FJ 12346789");
+    assertThat(detectedMgrsStrings, both(hasSize(1)).and(contains("4QFJ0123406789")));
+  }
+
+  @Test
+  public void testProcessorDetectsPrecision4NoSpaceAtAll() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4QFJ12346789");
+    assertThat(detectedMgrsStrings, both(hasSize(1)).and(contains("4QFJ0123406789")));
+  }
+
+  @Test
+  public void testProcessorDetectsPrecision5() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4Q FJ 12345 67891");
+    assertThat(detectedMgrsStrings, both(hasSize(1)).and(contains("4QFJ1234567891")));
+  }
+
+  @Test
+  public void testProcessorDetectsPrecision5OffsetLeft() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4Q FJ 123456 7891");
+    assertThat(
+        detectedMgrsStrings, both(hasSize(2)).and(contains("4QFJ0012300456", "4QFJ0007800091")));
+  }
+
+  @Test
+  public void testProcessorDetectsPrecision5OffsetRight() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4Q FJ 1234 567891");
+    assertThat(
+        detectedMgrsStrings, both(hasSize(2)).and(contains("4QFJ0001200034", "4QFJ0056700891")));
+  }
+
+  @Test
+  public void testProcessorDetectsPrecision5NoSpace() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4Q FJ 1234567891");
+    assertThat(detectedMgrsStrings, both(hasSize(1)).and(contains("4QFJ1234567891")));
+  }
+
+  @Test
+  public void testProcessorDetectsPrecision5NoSpaceAtAll() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4QFJ1234567891");
+    assertThat(detectedMgrsStrings, both(hasSize(1)).and(contains("4QFJ1234567891")));
+  }
+
+  @Test
+  public void testProcessorHandlesAttemptedPrecision6() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4Q FJ 123456 789123");
+    assertThat(
+        detectedMgrsStrings, both(hasSize(2)).and(contains("4QFJ0012300456", "4QFJ0078900123")));
+  }
+
+  @Test
+  public void testProcessorHandlesAttemptedPrecision7() {
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings("4Q FJ 1234567 8912345");
+    assertThat(
+        detectedMgrsStrings, both(hasSize(2)).and(contains("4QFJ0012300456", "4QFJ0089100234")));
+  }
+
+  @Test
+  public void testProcessorIgnoresDuplicateZone() {
+    String query = "1D 1D"; // Should yield 2 points
+    List<Suggestion> suggestions = processor.enhanceResults(new LinkedList<>(), query);
+    List<String> detectedMgrsStrings = processor.getMgrsCoordinateStrings(query);
+    assertThat(detectedMgrsStrings, both(hasSize(2)).and(contains("1D", "1D")));
```

Comment:
This test confused me. If you move the `.stream().distinct().collect(Collectors.toList())` to the return statement of `coordinateStringsFromQuery` couldn't you demonstrate that you eliminate duplicates in this first assert statement?